lib/validators/specValidator.js

```diff
@@ -568,7 +568,21 @@ class SpecValidator {
     let parameters = operation.getParameters();
     let result = { request: null, validationResult: { errors: [], warnings: [] } }, foundIssues = false;
     let options = {};
-
+    if (operation.pathObject && operation.pathObject.api && operation.pathObject.api.host) {
+      let scheme = 'https';
+      let basePath = '';
+      let host = operation.pathObject.api.host;
+      if (host.endsWith('/'))
+        host = host.slice(0, host.length - 1);
+      if (operation.pathObject.api.schemes && !operation.pathObject.api.schemes.some((item) => { return item && item.toLowerCase() === 'https'; }))
+        scheme = operation.pathObject.api.schemes[0];
```

Comment:
if this array has multiple, how do we know we should pick the first?

Reply:
i am just picking the first one only if the array does not contain 'https'. In that scenario we any ways do not have an option.